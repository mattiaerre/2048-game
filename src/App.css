.App {
  text-align: center;
}

.App__table {
  margin: auto;
}

.App__table tbody tr td {
  border: 1px solid black;
  font-family: monospace;
  font-size: 64px;
  height: 100px;
  width: 100px;
}
