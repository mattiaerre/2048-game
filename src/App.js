import React, { useEffect, useReducer } from 'react';
import { version } from '../package.json';
import { DOWN, LEFT, NEW_GAME, RIGHT, UP } from './core/constants';
import initialState from './store/initialState';
import reducer from './store/reducer';
import './App.css';
import useKeyPress from './useKeyPress';

function App() {
  const downPress = useKeyPress(40);
  const leftPress = useKeyPress(37);
  const rightPress = useKeyPress(39);
  const upPress = useKeyPress(38);

  const [state, dispatch] = useReducer(reducer, initialState);

  useEffect(() => {
    if (downPress) {
      dispatch({ type: DOWN });
    }
    if (leftPress) {
      dispatch({ type: LEFT });
    }
    if (rightPress) {
      dispatch({ type: RIGHT });
    }
    if (upPress) {
      dispatch({ type: UP });
    }
  }, [downPress, leftPress, rightPress, upPress]);

  return (
    <article className="App">
      <h1 className="App__h1">
        <a
          href="https://2048game.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          2048
        </a>
      </h1>
      <p>
        <button
          className="App__button"
          onClick={() => dispatch({ type: NEW_GAME })}
        >
          New game
        </button>
      </p>
      <table className="App__table">
        <tbody>
          {state.map((row, index) => {
            return (
              <tr key={index}>
                {row.map((cell, index) => (
                  <td className={`tile--${cell}`} key={index}>
                    {cell}
                  </td>
                ))}
              </tr>
            );
          })}
        </tbody>
      </table>
      <p className="App__p">
        How to play: use your arrow keys to move the tiles. When two tiles with
        the same number touch, they merge into one.
      </p>
      <footer className="App__footer">
        <small>v{version}</small>
      </footer>
    </article>
  );
}

export default App;
