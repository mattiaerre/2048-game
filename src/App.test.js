import React from 'react';
import { render } from '@testing-library/react';
import mockAddRandom from './core/mockAddRandom';
import App from './App';

jest.mock('./core/addRandom', () => (afterMove) => mockAddRandom(afterMove));

test('<App />', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
