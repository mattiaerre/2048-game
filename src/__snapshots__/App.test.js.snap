// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> 1`] = `
<div>
  <article
    class="App"
  >
    <h1
      class="App__h1"
    >
      <a
        href="https://2048game.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        2048
      </a>
    </h1>
    <p>
      <button
        class="App__button"
      >
        New game
      </button>
    </p>
    <table
      class="App__table"
    >
      <tbody>
        <tr>
          <td
            class="tile--null"
          />
          <td
            class="tile--null"
          />
          <td
            class="tile--null"
          />
          <td
            class="tile--null"
          />
        </tr>
        <tr>
          <td
            class="tile--null"
          />
          <td
            class="tile--null"
          />
          <td
            class="tile--null"
          />
          <td
            class="tile--null"
          />
        </tr>
        <tr>
          <td
            class="tile--null"
          />
          <td
            class="tile--null"
          />
          <td
            class="tile--null"
          />
          <td
            class="tile--null"
          />
        </tr>
        <tr>
          <td
            class="tile--null"
          />
          <td
            class="tile--null"
          />
          <td
            class="tile--null"
          />
          <td
            class="tile--2"
          >
            2
          </td>
        </tr>
      </tbody>
    </table>
    <p
      class="App__p"
    >
      How to play: use your arrow keys to move the tiles. When two tiles with the same number touch, they merge into one.
    </p>
    <footer
      class="App__footer"
    >
      <small>
        v
        0.1.4
      </small>
    </footer>
  </article>
</div>
`;
